addons/l10n_no_edi/models/account_edi_format.py

```diff
@@ -0,0 +1,121 @@
+# -*- coding: utf-8 -*-
+
+from odoo import models, _
+from odoo.addons.account_edi_ubl_bis3.models.account_edi_format import COUNTRY_EAS
+
+import base64
+import markupsafe
+
+
+class AccountEdiFormat(models.Model):
+    _inherit = 'account.edi.format'
+
+    ####################################################
+    # Import
+    ####################################################
+
+    def _is_ubl(self, filename, tree):
+        """ OVERRIDE so that the generic ubl parser does not parse BIS3 any longer.
+        """
+        is_ubl = super()._is_ubl(filename, tree)
+        return is_ubl and not self._is_ehf_3(filename, tree)
+
+    def _is_ehf_3(self, filename, tree):
+        profile_id = tree.find('./{*}ProfileID')
+        nor_tax = tree.xpath("//*[text()='Foretaksregisteret']")
```

Comment:
Is this really the only way to detect it?

Incidentally it seems like a waste to create those locals as it's going to perform those sub-searches even if the root is not an UBL invoice.

Also for the profile `findtext` could be better than `find` then dereferencing the `text`.

Reply:
Foretaksregisteret is basically the only difference for the Norwegian profile of BIS3, so yes, it's the only way.
I added the namespaces and better refined the path for nor_tax.
I use `findtext` for ProfileID.
The use of `findtext` is difficult for nor_tax has the same node is present twice and we need the one with the correct text.